tensorflow/lite/micro/kernels/batch_matmul_test_util.h

```diff
@@ -0,0 +1,345 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_LITE_MICRO_KERNELS_BATCH_MATMUL_TEST_UTIL_H_
+#define TENSORFLOW_LITE_MICRO_KERNELS_BATCH_MATMUL_TEST_UTIL_H_
+
+#include <cstdint>
+#include <initializer_list>
+#include <type_traits>
+
+#include "tensorflow/lite/c/builtin_op_data.h"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/tensor_utils_common.h"
+#include "tensorflow/lite/micro/kernels/kernel_runner.h"
+#include "tensorflow/lite/micro/test_helpers.h"
+#include "tensorflow/lite/micro/testing/micro_test.h"
+
+namespace tflite {
+namespace testing {
+
+constexpr int kMaxDims = RuntimeShape::kMaxSmallSize;
+constexpr int kInputTensor1 = 0;
+constexpr int kInputTensor2 = 1;
+constexpr int kOutputTensor = 2;
+constexpr float kTolerance = 1e-5;
+
+enum TestInputIndex {
+  kInputIndex0,
+  kInputIndex1,
+};
+constexpr size_t kMaxInputs = TestInputIndex::kInputIndex1 + 1;
+
+struct TestTensorData {
```

Comment:
I'm guessing that the code here is a prototype of a more general set of testing utilities, and will be moved into a less op-specific header in the future if we like this direction? Maybe a comment to that effect, and a link to docs or bugs (if any) about this work would be useful?

Reply:
This is generic code.  Should the TFLM team decide to use this style of test code, I would expect the file to be renamed.